src/compiler/parser.ts

```diff
@@ -9444,15 +9444,13 @@ namespace Parser {
             }
 
             function parseSeeTag(start: number, tagName: Identifier, indent?: number, indentText?: string): JSDocSeeTag {
-                const isMarkdownOrJSDocLink = token() === SyntaxKind.OpenBracketToken
-                    || lookAhead(() => nextTokenJSDoc() === SyntaxKind.AtToken && tokenIsIdentifierOrKeyword(nextTokenJSDoc()) && isJSDocLinkTag(scanner.getTokenValue()));
-                const nameExpression = isMarkdownOrJSDocLink ? undefined : parseJSDocNameReference();
+                const nameExpression = isMarkdownOrJSDocLink() ? undefined : parseJSDocNameReference();
                 const comments = indent !== undefined && indentText !== undefined ? parseTrailingTagComments(start, getNodePos(), indent, indentText) : undefined;
                 return finishNode(factory.createJSDocSeeTag(tagName, nameExpression, comments), start);
             }
 
             function parseThrowsTag(start: number, tagName: Identifier, indent: number, indentText: string): JSDocThrowsTag {

```

Comment:
@LukeAbby suggested `@returns` and `@yields` could use the same treatment. `@yields` isn't a specialized tag in TS today so I skipped that and I'm slightly on the fence when it comes to `@returns`. cc @sandersn